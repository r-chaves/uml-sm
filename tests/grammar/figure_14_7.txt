stm uml_figure_14_7
{
    state Dialing
    {
        initial -> Start

        state Start
        {
            entry / start dial tone
            exit  / stop dial tone
        }

        Start -> Partial_Dial : digit(n)

        state Partial_Dial
        {
            entry / number.append(n)
        }

        Partial_Dial -> Partial_Dial : digit(n)
        Partial_Dial -> final ? [number.isValid()]
    }
}
