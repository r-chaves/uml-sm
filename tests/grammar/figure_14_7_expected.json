{
  "stm_name": "uml_figure_14_7", 
  "region_member": {
    "region_member": [
      {
        "src": "initial", 
        "guard": null, 
        "trigger": null, 
        "target": {
          "scope": null, 
          "region": null, 
          "conn_point": null, 
          "name": "Start"
        }, 
        "effect": null
      }, 
      {
        "region_member": null, 
        "region": null, 
        "name": "Start", 
        "behavior": [
          {
            "kind": "entry", 
            "effect": "start dial tone"
          }, 
          {
            "kind": "exit", 
            "effect": "stop dial tone"
          }
        ]
      }, 
      {
        "src": {
          "scope": null, 
          "region": null, 
          "conn_point": null, 
          "name": "Start"
        }, 
        "guard": null, 
        "trigger": {
          "name": "digit(n)"
        }, 
        "target": {
          "scope": null, 
          "region": null, 
          "conn_point": null, 
          "name": "Partial_Dial"
        }, 
        "effect": null
      }, 
      {
        "region_member": null, 
        "region": null, 
        "name": "Partial_Dial", 
        "behavior": {
          "kind": "entry", 
          "effect": "number.append(n)"
        }
      }, 
      {
        "src": {
          "scope": null, 
          "region": null, 
          "conn_point": null, 
          "name": "Partial_Dial"
        }, 
        "guard": null, 
        "trigger": {
          "name": "digit(n)"
        }, 
        "target": {
          "scope": null, 
          "region": null, 
          "conn_point": null, 
          "name": "Partial_Dial"
        }, 
        "effect": null
      }, 
      {
        "src": "Partial_Dial", 
        "guard": "number.isValid()", 
        "trigger": null, 
        "target": "final", 
        "effect": null
      }
    ], 
    "region": null, 
    "name": "Dialing", 
    "behavior": null
  }
}
